Src/Newtonsoft.Json.Tests/Serialization/JsonSerializerTest.cs

```diff
@@ -8076,6 +8076,17 @@ public void SetMaxDepth_MaxValueDepthNotExceeded()
             Assert.AreEqual(150, depth);
         }
 
+        [Test]
+        public void ShallowCopy_DoesntSetDateFormatStringSet()
+        {
+            var settings = new JsonSerializerSettings();
+
+            var clone = new JsonSerializerSettings(settings);
+
+            Assert.AreEqual(false, settings._dateFormatStringSet);
+            Assert.AreEqual(false, clone._dateFormatStringSet);
```

Comment:
Can we use reflection to verify all fields/properties are copied? Similar to [this CloneSslOptionsClonesAllProperties test](https://github.com/dotnet/aspnetcore/blob/8b30d862de6c9146f466061d51aa3f1414ee2337/src/Servers/Kestrel/Core/test/SniOptionsSelectorTests.cs#L658-L749) in Kestrel. The basic idea is to use GetFields()/GetProperties() to get every property and then remove each property from the list after verifying that a non-default value was copied. Then at the end of the test you verify the list is empty. The test will need updated any time a property is added, but at least the test should fail if a property is added without being copied by the constructor.

Reply:
@halter73 Done (better late than never... :))